bpf/lbmap/lb.go

```diff
@@ -435,129 +451,145 @@ func lbLookupKey(ctx *cli.Context) {
 			return
 		}
 
-		fmt.Printf("key.vip %v, key.dport %d\n", key.vip, key.dport)
-		fmt.Printf("value.state %d\n", lbval.state)
+		fmt.Printf("key.address %v, key.dport %d\n",
+			key.address, common.Swab16(key.dport))
+		fmt.Printf("%+v\n", lbval)
 	} else if lbtype == lbState {
-		key, err := strconv.ParseUint(ctx.Args().Get(2), 10, 16)
+		key, err := strconv.ParseUint(ctx.Args().Get(2), 0, 16)
 		if err != nil {
 			printArgsUsageAndExit(ctx)
 			return
 		}
 
-		lbval, err := lookupLb2(file, uint16(key))
+		lbval, err := lookupLb2(file, common.Swab16(uint16(key)))
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "%s\n", err)
 			os.Exit(errIOFailure)
 			return
 		}
 
 		fmt.Printf("key.dport %d\n", uint16(key))
-		fmt.Printf("value.state %d\n", lbval.state)
+		fmt.Printf("%+v\n", lbval)
 	} else {
 		fmt.Fprintf(os.Stderr, "Incorrect lbtype %d.\n", lbtype)
 		printArgsUsageAndExit(ctx)
 		return
 	}
 }
 
-func lbUpdateKey(ctx *cli.Context) {
-	lbval := LBValue{}
+func lbUpdateService(ctx *cli.Context) {
+	lbval := LBService{}
+	key := LBKey{}
 
-	if len(ctx.Args()) < 5 {
+	if len(ctx.Args()) < 6 {
 		printArgsUsageAndExit(ctx)
 		return
 	}
 
-	file := ctx.Args().Get(0)
-
-	lbtype, err := strconv.ParseUint(ctx.Args().Get(1), 10, 8)
+	tmp, err := strconv.ParseUint(ctx.Args().Get(0), 0, 16)
 	if err != nil {
+		fmt.Fprintf(os.Stderr, "%s\n", err)
 		printArgsUsageAndExit(ctx)
 		return
 	}
+	key.slave = uint16(tmp)
 
-	iv6 := net.ParseIP(ctx.Args().Get(2))
+	iv6 := net.ParseIP(ctx.Args().Get(1))
 	if len(iv6) != net.IPv6len {
 		fmt.Fprintf(os.Stderr, "invalid IPv6\n")
 		printErrorAndExit(ctx, "invalid IPv6\n", errInvalidArgument)
 		return
 	}
-	copy(lbval.vip[:], iv6)
+	copy(key.address[:], iv6)
 
-	tmp, err := strconv.ParseUint(ctx.Args().Get(3), 10, 16)
+	tmp, err = strconv.ParseUint(ctx.Args().Get(2), 0, 16)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "%s\n", err)
 		printArgsUsageAndExit(ctx)
 		return
 	}
-	lbval.dport = uint16(tmp)
+	lbval.port = common.Swab16(uint16(tmp))
+	key.dport = lbval.port
 
-	tmp, err = strconv.ParseUint(ctx.Args().Get(4), 10, 16)
+	tmp, err = strconv.ParseUint(ctx.Args().Get(3), 0, 16)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "%s\n", err)
 		printArgsUsageAndExit(ctx)
 		return
 	}
-	lbval.state = uint16(tmp)
+	state := uint16(tmp)
 
-	count, err := strconv.ParseUint(ctx.Args().Get(5), 10, 16)
-	if err != nil || count >= MaxLxc {
+	count, err := strconv.ParseUint(ctx.Args().Get(4), 0, 16)
+	if err != nil {
 		fmt.Fprintf(os.Stderr, "%s\n", err)
 		printArgsUsageAndExit(ctx)
 		return
 	}
-	lbval.lxcCount = int32(count)
+	lbval.count = uint16(count)
```

Comment:
I suppose the count is only needed for the master and not the slaves. So should we update the map count, only the master changes with the addition or deletion of a slave.